src/libasr/codegen/wasm_to_x86.cpp

```diff
@@ -129,8 +129,38 @@ class X86Visitor : public WASMDecoder<X86Visitor>,
 
     void visit_EmtpyBlockType() {}
 
+    void visit_Br(uint32_t /*label_index*/) {
+        // Branch is used to jump to the `loop.head`.
+        m_a.asm_jmp_label(".loop.head_" + unique_id.rbegin()[1]);
+    }
+
```

Comment:
What is the use of `branch`?
My doubt is:
1. the label_index is 1 if the visitor is either `break` or `end` of the `if.block`
2. the label_index is 2 if the visitor is `continue`
if.block and continue must have the same behavior, right?

Reply:
I see, the label_index is not constant, it changes based on the statements inside the loop.

Reply:
>What is the use of branch?

It is like a `jump` instruction. Everytime we wish to continue a `loop` in `wasm`, we need to `branch` (only if the `loop` condition is `true`) to the `loop`.

https://samrat.me/posts/2020-03-29-webassembly-control-instr-examples/ offers a good explanation.

Reply:
>I see, the label_index is not constant, it changes based on the statements inside the loop.

Yes, the `label_index` is used as the target of the `branch`.

https://github.com/lcompilers/lpython/blob/ce4612d5d2a7207ab8147bb102e39bbbe5e67e2c/src/libasr/codegen/asr_to_wasm.cpp#L2025-L2029

As shared in the above comment, the label index counts from the inner most `block` (A `block` could be a `loop` or an `if-else` construct). In our case, a `label_index` of `0` points to the `if-else` condition (which is the loop condition) and a `label_index` of `1` points to the actual `loop`.

Also, on branching to a `loop`, the control passes to the beginning of the `loop`, where as on branching to an `if` block, the control passes to the end of it.

Reply:
>// Unlike with other index spaces, indexing of labels is relative by 
 // nesting depth, that is, label 0 refers to the innermost structured 
 // control instruction enclosing the referring branch instruction, while 
 // increasing indices refer to those farther out. 

The following example should (hopefully) explain the above more clearly.


```
// Code in some dummy language, (might) not be exactly in `WebAssembly Text Format`

(loop // label index = 3
    (if // label index = 2
        (if // label index = 1
            (if // label index = 0
                br some_label_index
            )
        )
    )
) 
```
Depending on where we would like to branch, we can have appropriate value for `some_label_index`.

Reply:
I got it, great explanation. Thanks